lux/extensions/auth/acl/__init__.py

```diff
@@ -0,0 +1,32 @@
+
+
+def acl_ids_query(request, session, user_id,
+                  resource, action, effect='allow', id_type=None):
+    """
+    Note: initially res_ids are of string type
+    :param request: pulsar wsgi request object
+    :param session: sqlalchemy session
+    :param user_id: id from model "user"
+    :param resource: arbitrary name (alias), not linked to any data
+    :param action: read/update/create
+    :param effect: allow or deny
+    :param id_type: the type res_id has to be cast to
+    :return:
+    True if no restrictions (i.e. res_id == '*'
+        in one of records found);
+    Query for limited number of res_ids or no res_ids
+
+    """
+    odm = request.app.odm()
+    q = session.query(odm.acl.res_id)
+    if q.filter(odm.acl.res_id == '*').count():
```

Comment:
What about if the `effect` in the policy is `deny`? Also, `.first()` is more efficient.